### ISSUE TITLE
CA1849 do not warn for EF AddRangeAsync

### ISSUE THREAD

**Issue by molinch** [dotnet/roslyn-analyzers#6718]
### Analyzer

Diagnostic ID: [CA1849](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1849) Call async methods when in an async method

### Describe the improvement

Right now CA1849 warns if EntityFramework Core AddRange is used in an async function. It complains because AddRangeAsync could be used. It is wrong, if you don't use db generated values then you should never use AddAsync/AddRangeAsync methods. For reference this is the official method documentation:
```
/// This method is async only to allow special value generators, such as the one used by
/// 'Microsoft.EntityFrameworkCore.Metadata.SqlServerValueGenerationStrategy.SequenceHiLo',
/// to access the database asynchronously. For all other cases the non async method should be used.
```

### Describe suggestions on how to achieve the rule

CA1849 should not warn when using AddRange in an async function.

**Comment by CollinAlpert** [comment]
Duplicate of #6684.
